app/src/main/java/org/mozilla/tv/firefox/webrender/WebViewCache.kt

```diff
@@ -64,10 +67,15 @@ class WebViewCache : LifecycleObserver {
             return SystemEngineView(context, attrs).apply {
                 state?.let { this.restoreState(it) }
                 initialize()
-            }.also { cachedView = it }
+            }.also {
+                cachedView = it
+                doAfterRecreate?.invoke()
+                doAfterRecreate = null
+            }
         }
 
         cachedView?.removeFromParentIfAble()
+        sessionRepo.backForwardIndexProvider = { cachedView?.currentBackForwardIndex() ?: -1 }
```

Comment:
What does this do? Why is it necessary? Add a comment, probably on the definition in the SessionRepo.